### PR TITLE
Hotfix - NPE crash in RouterServiceProvider

### LINKED ISSUE #365: NPE in SdlRouterStatusProvider
Fatal Exception: java.lang.NullPointerException: Attempt to read from field 'eeq eeo.c' on a null object reference
       at com.smartdevicelink.transport.SdlRouterStatusProvider$ClientHandler.com.smartdevicelink.transport.SdlRouterStatusProvider.handleRouterStatusConnectedResponse(SourceFile:1112)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5343)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)

### PR DESCRIPTION
**Pull request by joeygrover**
Fixes #365 

Also fixed a few other obvious issues along the same line in the SdlRouterService.

### CODE REVIEW

File: sdl_android_lib/src/com/smartdevicelink/transport/SdlRouterService.java [hunk 1]
```diff
@@ -328,6 +328,9 @@ public RouterHandler(SdlRouterService provider){
 	    	
 	        @Override
 	        public void handleMessage(Message msg) {
+	        	if(this.provider == null || this.provider.get() == null){
```
Comment (valeriia-shchukina) [inline sdl_android_lib/src/com/smartdevicelink/transport/SdlRouterService.java]: Could make this.provider variable final, initialize it once in constructor and not to check for (this.provider == null) each time after. Only this.provider.get() can actually be null.

Comment (joeygrover) [inline sdl_android_lib/src/com/smartdevicelink/transport/SdlRouterService.java]: Yes that's a better idea

**Comment by valeriia-shchukina** [comment]
seems good) waiting for the fix to be merged

**Comment by valeriia-shchukina** [comment]
Can we ship it, please?? It's really urgent, ton of crashes in production =(

**Comment by mrapitis** [comment]
LGTM

**Pull request merged**